### PR TITLE
Architecture details of Broker cluster

### PR DESCRIPTION
**Pull request by sridhargaddam**
Related to: https://github.com/submariner-io/submariner-website/issues/18

Signed-off-by: Sridhar Gaddam <sgaddam@redhat.com>

### CODE REVIEW

**Comment by tpantelis** [comment]
@sridhargaddam I've been working on the Architecture sections, including Broker, with https://github.com/submariner-io/submariner-website/pull/134. Is it OK if I look over what you added and incorporate into mine?

File: src/content/architecture/broker/_index.en.md [hunk 1]
```diff
@@ -2,9 +2,28 @@
...
+
+It is to be noted that Submariner does not run any Pods on the Broker cluster. It only uses it as a datastore for sync'ing information of all the connected clusters.
+On the Broker cluster, the information associated with all the data clusters is stored as cluster and endpoint CRDs in `submariner-k8s-broker` namespace.
+When a data cluster is joined to the Broker, the `submariner-gateway` Pod on the data cluster will push the local cluster and endpoint CRD info to the Broker and will sync information of other clusters (based on the cluster ID) from the Broker to the local cluster.
+This information will be used by the data clusters to setup tunnels to other clusters.
+
+![Figure 1 - Problem with overlapping CIDRs](/images/brokercluster.png)
+
+{{% notice info %}}
+Currently, the broker is implemented by utilizing the Kubernetes API, but is modular and can be enhanced in the future to bring support for other interfaces.
```
Comment (tpantelis) [inline src/content/architecture/broker/_index.en.md:21]: This is outlined in the Gateway Engine section.

File: src/content/architecture/broker/_index.en.md [hunk 2]
```diff
@@ -2,9 +2,28 @@
...
+{{% /notice %}}
+
+To query the list of clusters and endpoints stored on the broker, you can run the following commands
+
+```bash
+kubectl --kubeconfig <kubeConfigOfBrokerCluster> get clusters -n submariner-k8s-broker
+kubectl --kubeconfig <kubeConfigOfBrokerCluster> get endpoint -n submariner-k8s-broker
```
Comment (tpantelis) [inline src/content/architecture/broker/_index.en.md:28]: I don't think this belongs in an architecture section - this seems more related to deployment or troubleshooting.

File: src/content/architecture/broker/_index.en.md [hunk 3]
```diff
@@ -2,9 +2,28 @@
...
+
+{{% notice tip %}}
+
+Broker can also be deployed on a standalone kubernetes cluster which is not part of Submariner data cluster fleet.
+
+{{% /notice %}}
+
+It is to be noted that Submariner does not run any Pods on the Broker cluster. It only uses it as a datastore for sync'ing information of all the connected clusters.
+On the Broker cluster, the information associated with all the data clusters is stored as cluster and endpoint CRDs in `submariner-k8s-broker` namespace.
+When a data cluster is joined to the Broker, the `submariner-gateway` Pod on the data cluster will push the local cluster and endpoint CRD info to the Broker and will sync information of other clusters (based on the cluster ID) from the Broker to the local cluster.
```
Comment (tpantelis) [inline src/content/architecture/broker/_index.en.md:15]: This is outlined in the Gateway Engine section.

**Comment by tpantelis** [comment]
> @sridhargaddam I've been working on the Architecture sections, including Broker, with #134. Is it OK if I look over what you added and incorporate into mine?

I pushed https://github.com/submariner-io/submariner-website/pull/134/commits/17eb6c8d121bed8b11828bad4093de3c381265e9 to update the Broker section with some text from here that I though was relevant.  I'm not sure about the intent of the new diagram - perhaps that can still be proposed with this PR.

**Comment by sridhargaddam** [comment]
> > @sridhargaddam I've been working on the Architecture sections, including Broker, with #134. Is it OK if I look over what you added and incorporate into mine?
> 
> I pushed [17eb6c8](https://github.com/submariner-io/submariner-website/commit/17eb6c8d121bed8b11828bad4093de3c381265e9) to update the Broker section with some text from here that I though was relevant. I'm not sure about the intent of the new diagram - perhaps that can still be proposed with this PR.

I was not aware that you pushed a PR for architecture. I'll abandon this PR and we can continue reviewing your PR. It looks great, thanks.

**Comment by sridhargaddam** [comment]
> @sridhargaddam I've been working on the Architecture sections, including Broker, with #134. Is it OK if I look over what you added and incorporate into mine?

Sure, please go ahead.

**Comment by sridhargaddam** [comment]
> With #134, does this still require review?

I'm closing it.

**Pull request closed without merging**